### PR TITLE
feat: buy now creation

### PR DESCRIPTION
**Pull request by jmulq**
This PR includes buy now creation logic, and updating the datasource to mongodb.

There is a subgraph PR that goes alongside this - https://github.com/Vault-Hill/vhc-marketplace-subgraph/pull/9

### CODE REVIEW

File: src/graphql-types/asset.ts [hunk 1]
```diff
@@ -30,6 +34,13 @@ export class Asset extends BaseEntity {
   @property()
   owner: string
 
+  // TODO: Add Auction type when implemented
```
Comment (anjorinjnr) [inline src/graphql-types/asset.ts:37]: do you intend to add a separate property for auctions here?

File: src/repositories/buy-now.repository.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  async getAll() {
+    return this.find()
+  }
+
+  async getBuyNow(assetId: string, assetAddress: string) {
+    return this.findOne({ where: { assetAddress, assetId } })
+  }
+
+  async add(data: Order721Input) {
+    const buyNow: Partial<BuyNow> = {
```
Comment (anjorinjnr) [inline src/repositories/buy-now.repository.ts:25]: can there be muutiple active buynow orders for an asset?

**Review by anjorinjnr — CHANGES_REQUESTED** [review]
this is look mostly good.

since there's no unit/integration test yet, have you provisioned a test/staging db and did you test with that?

File: src/graphql-types/asset.ts (hunk 1 above)
Comment (jmulq) [reply src/graphql-types/asset.ts:37]: My plan was to make the type be one of `BuyNow` or `Auction`. That way there can only be on active listing, that is one of those two types.

File: src/repositories/buy-now.repository.ts (hunk 2 above)
Comment (jmulq) [reply src/repositories/buy-now.repository.ts:25]: No. Once a buy now has been created for an asset, no further buy nows should be able to be created with our solution.

Comment (anjorinjnr) [reply src/repositories/buy-now.repository.ts:25]: that's what I thought so you should probably add a validation here.

File: src/graphql-types/asset.ts (hunk 1 above)
Comment (anjorinjnr) [reply src/graphql-types/asset.ts:37]: Ok. So to resolve the field, you will have to run two queries BuyNow and Auction? Perhaps you can use a single collection to represent both?

**Review by anjorinjnr — COMMENTED** [review]
See comments

File: src/repositories/buy-now.repository.ts (hunk 2 above)
Comment (jmulq) [reply src/repositories/buy-now.repository.ts]: Added validation to the resolver. See latest commit

File: src/graphql-types/asset.ts (hunk 1 above)
Comment (jmulq) [reply src/graphql-types/asset.ts]: Have created a `Listing` entity now that will represent both. See latest commit.

Comment (anjorinjnr) [reply src/graphql-types/asset.ts]: lgtm

**Review by anjorinjnr — CHANGES_REQUESTED** [review]
lgtm

please run lint:fix and squash the commits.

File: src/graphql-types/asset.ts [hunk 3]
```diff
@@ -30,6 +34,12 @@ export class Asset extends BaseEntity {
   @property()
   owner: string
 
+  @field((type) => Listing, {
+    nullable: true
+  })
+  @property({ type: () => Listing })
+  activeListing?: Listing
```
Comment (anjorinjnr) [inline src/graphql-types/asset.ts:41]: should this just be `listing`? the associated listing may not always be active right?

**Review by anjorinjnr — APPROVED** [review]
lgtm see minot comment

**Comment by anjorinjnr** [comment]
Also pls add dev values in the .env example, so someone can actually checkout the code and run it.

File: src/graphql-types/asset.ts (hunk 3 above)
Comment (jmulq) [reply src/graphql-types/asset.ts:41]: I was thinking that an associated `Listing` would always be active. i.e. when a listing is cancelled or finished, we set this field to null.

**Review by anjorinjnr — DISMISSED** [review]

**Pull request merged**